### ISSUE TITLE
App service creation not working

### ISSUE THREAD

**Issue by jkubrynski** [Azure/azure-sdk-for-java#1174]
I was trying to use `WebSiteManagementClientImpl` to create `sites` object. And it looks there is a bug in the backend API. Documentation (and Java SDK implementation) tells that properties like `alwaysOn`, `webSocketsEnabled` or `defaultDocuments` should be set on `properties` node of the `siteConfig` object. That's not working at all, but when I set those values directly on `siteConfig` it works correctly. Below you can find correct document, and document generated by SDK. It's generally not your bug, but I have no idea where can I submit bug in backend API :)

SDK:

```
{
    "location": "westeurope",
    "properties": {
        "name": "MY_SITE_NAME",
        "serverFarmId": "MY_FARM_ID",
        "clientAffinityEnabled":false,
        "siteConfig": {
            "location": "westeurope",
            "properties": {
                    "alwaysOn":true,
                    "webSocketsEnabled":true,
                    "defaultDocuments":["index.html"]
            }  
        }
    }
}
```

Working document:

```
{
    "location": "westeurope",
    "properties": {
        "name": "MY_SITE_NAME",
        "serverFarmId": "MY_FARM_ID",
        "clientAffinityEnabled":false,
        "siteConfig": {
            "location": "westeurope",
            "alwaysOn":true,
            "webSocketsEnabled":true,
            "defaultDocuments":["index.html"]
        }
    }
}
```

**Comment by martinsawicki** [comment]
smells like a Swagger issue or maybe at the REST level. We haven't looked at this in depth yet, but @jianghaolu has been the closest so maybe he has some input already or knows who to pass it on to.

**Comment by jianghaolu** [comment]
Planning to take a look at this sometime this month. Meanwhile, https://github.com/azure/azure-rest-api-specs is the place where backend service owners shop around for issues.

**Comment by jkubrynski** [comment]
Thanks @jianghaolu - I've submitted https://github.com/Azure/azure-rest-api-specs/issues/589

**Comment by amarzavery** [comment]
@jianghaolu - The siteconfig properties are flattened in the sdk as per the x-ms-client-flatten extension, which is the right thing.

Intellisense should guide the customer to do the right thing.

**Comment by jkubrynski** [comment]
@amarzavery I'm using the SDK and the results is all those properties are ignored by Azure. For example I see that there is "alwaysOn" flag being set in the request body, however finally the created site has alwaysOn set to false.

**Comment by jianghaolu** [comment]
Thanks @amarzavery for answering. This is correct, The SDK should be able to move the properties under `properties` property.
So now we just need to figure out if SDK did it or not.
@jkubrynski  Can you turn on logging and see what's the payload? You can turn on logging from `OkHttp.Builder` which is accepted by the `WebsiteManagmentClientImpl` constructor.

**Comment by jkubrynski** [comment]
The document in the first comment is produced by SDK. Do you need something more?

**Comment by jianghaolu** [comment]
@jkubrynski Then it's an SDK bug. I'll close the issue in the rest-api-specs and investigate it here.

**Comment by jianghaolu** [comment]
@jkubrynski This should be fixed in 1.0 of the SDK. The `SiteConfig` class has all the properties flattened: https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/SiteConfig.java.
Thanks!
